pkgs/tools/virtualization/google-compute-engine/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, fetchFromGitHub
+, pythonPackages
+, hwclock
+, shadow
+, systemd
+, utillinux
+}:
+let
+  version = "20170523";
+in
+pythonPackages.buildPythonApplication {
+  name = "google-compute-engine-${version}";
+  namePrefix = "";
+
+  src = fetchFromGitHub {
+    owner = "GoogleCloudPlatform";
+    repo = "compute-image-packages";
+    rev = version;
+    sha256 = "1qxyj3lj9in6m8yi6y6wcmc3662h9z4qax07v97rdnay99mxdv68";
+  };
+
+  patches = [ ./0001-allow-nologin-other-paths.patch ];
+
+  patchPhase = ''
```

Comment:
use `postPatch`, otherwise your `patches` is ignored because you define a new `patchPhase`.